tensorflow/tools/ci_build/linux/mkl/install_openmpi_horovod.sh

```diff
@@ -54,27 +56,52 @@ echo 'OpenMPI version:'
 mpirun --version
 
 # Install OpenSSH for MPI to communicate between containers
-apt-get clean && apt-get update && \
-    apt-get install -y --no-install-recommends --fix-missing \
-        openssh-client openssh-server libnuma-dev && \
-    rm -rf /var/lib/apt/lists/*
-if [[ $?  == "0" ]]; then
-    echo "PASS: OpenSSH installation"
+if [[ ${BUILD_SSH} == "yes" ]]; then
+	mkdir /tmp/buildssh
+	cd /tmp/buildssh && curl -fSsL -O http://www.zlib.net/zlib-1.2.11.tar.gz && tar -xzvf zlib-1.2.11.tar.gz && \
+		cd /tmp/buildssh/zlib-1.2.11 && ./configure && make && make install
+	cd /tmp/buildssh && curl -fSsL -O https://www.openssl.org/source/openssl-1.1.1.tar.gz && tar -xzvf openssl-1.1.1.tar.gz && \
+		cd  /tmp/buildssh/openssl-1.1.1 && ./config && make  && make test  && make install
+	cd /tmp/buildssh && curl -fSsL -O https://mirrors.sonic.net/pub/OpenBSD/OpenSSH/portable/openssh-8.4p1.tar.gz && \
+		tar -xzvf openssh-8.4p1.tar.gz && cd /tmp/buildssh/openssh-8.4p1 && \
+		./configure --with-md5-passwords  && make && \
+		groupadd sshd && useradd -M -g sshd -c 'sshd privsep' -d /var/empty -s /sbin/nologin sshd && passwd -l sshd && \
+		make install
+	apt-get clean && apt-get update && \
+	    apt-get install -y --no-install-recommends --fix-missing \
+	        libnuma-dev cmake 
+	grep -v StrictHostKeyChecking /usr/local/etc/ssh_config > /usr/local/etc/ssh_config.new
+        # Allow OpenSSH to talk to containers without asking for confirmation
+        echo " StrictHostKeyChecking no" >> /usr/local/etc/ssh_config.new
+        mv /usr/local/etc/ssh_config.new /usr/local/etc/ssh_config
 else
-    yum -y update && yum -y install numactl-devel openssh-server openssh-clients && \
-        yum clean all
-    if [[ $?  == "0" ]]; then
-        echo "PASS: OpenSSH installation"
-    else
-        echo "Unsupported Linux distribution. Aborting!" && exit 1
-    fi
+	apt-get clean && apt-get update && \
+	    apt-get install -y --no-install-recommends --fix-missing \
+	        openssh-client openssh-server libnuma-dev cmake && \
+	    rm -rf /var/lib/apt/lists/*
+	if [[ $?  == "0" ]]; then
+	    echo "PASS: OpenSSH installation"
+	else
+	    yum -y update && yum -y install numactl-devel openssh-server openssh-clients cmake && \
+	        yum clean all
+	    if [[ $?  == "0" ]]; then
+	        echo "PASS: OpenSSH installation"
+	    else
+	        echo "Unsupported Linux distribution. Aborting!" && exit 1
+	    fi
+	fi
+	grep -v StrictHostKeyChecking /etc/ssh/ssh_config > /etc/ssh/ssh_config.new
+        # Allow OpenSSH to talk to containers without asking for confirmation
+        echo " StrictHostKeyChecking no" >> /etc/ssh/ssh_config.new
+        mv /etc/ssh/ssh_config.new /etc/ssh/ssh_config
```

Comment:
This is the same for both `if [[ ${BUILD_SSH} == "yes" ]]` and `else`. Can we move it back out of the `if`?

Reply:
the path of locally built ssh_config is in /usr/local/etc/ssh_config compared to the package installation one which is in /etc/ssh/ssh_config

Reply:
I see. In that case, can we make a variable to store the path in the `if/else`, refactor the code out of `if/else`, and use that variable?

Reply:
ok changing it